File: app/src/main/java/br/com/nicetuber/base/BaseInteractor.java
package br.com.nicetuber.base;

import javax.inject.Inject;

import br.com.nicetuber.model.api.YoutubeApi;

/**
 * Created by thiagozg on 14/10/2017.
 */

public class BaseInteractor {

    @Inject
    protected YoutubeApi youtubeApi;

}

File: app/src/main/java/br/com/nicetuber/base/BaseInteractor.kt
package br.com.nicetuber.base

import br.com.nicetuber.model.api.YoutubeApi
import javax.inject.Inject

/**
 * Created by thiagozg on 14/10/2017.
 */

open class BaseInteractor {

    @Inject
    protected lateinit var youtubeApi: YoutubeApi

}
